cognitive/src/test/scala/com/microsoft/azure/synapse/ml/cognitive/split1/MultivariateAnamolyDetectionSuite.scala

```diff
@@ -0,0 +1,193 @@
+import com.microsoft.azure.synapse.ml.cognitive.split1.MADListModelsProtocol._
+
+object MADUtils extends AnomalyKey {
+
+  import com.microsoft.azure.synapse.ml.cognitive.RESTHelpers._
+
+  def madSend(request: HttpRequestBase, path: String,
+              params: Map[String, String] = Map()): String = {
+
+    val paramString = if (params.isEmpty) {
+      ""
+    } else {
+      "?" + URLEncodingUtils.format(params)
+    }
+    request.setURI(new URI(path + paramString))
+
+    retry(List(100, 500, 1000), { () =>
+      request.addHeader("Ocp-Apim-Subscription-Key", anomalyKey)
+      request.addHeader("Content-Type", "application/json")
+      using(Client.execute(request)) { response =>
+        if (!response.getStatusLine.getStatusCode.toString.startsWith("2")) {
+          val bodyOpt = request match {
+            case er: HttpEntityEnclosingRequestBase => IOUtils.toString(er.getEntity.getContent, "UTF-8")
+            case _ => ""
+          }
+          if (response.getStatusLine.getStatusCode.toString.equals("429")) {
+            val retryTime = response.getHeaders("Retry-After").head.getValue.toInt * 1000
+            Thread.sleep(retryTime.toLong)
+          }
+          throw new RuntimeException(s"Failed: response: $response " + s"requestUrl: ${request.getURI}" +
+            s"requestBody: $bodyOpt")
+        }
+        if (response.getStatusLine.getReasonPhrase == "No Content") {
+          ""
+        }
+        else if (response.getStatusLine.getReasonPhrase == "Created") {
+          response.getHeaders("Location").head.getValue
+        }
+        else {
+          IOUtils.toString(response.getEntity.getContent, "UTF-8")
+        }
+      }.get
+    })
+  }
+
+  def madDelete(path: String, params: Map[String, String] = Map()): String = {
+    madSend(new HttpDelete(), "https://westus2.api.cognitive.microsoft.com/anomalydetector/" +
+      "v1.1-preview/multivariate/models/" + path, params)
+  }
+
+  def madListModels(params: Map[String, String] = Map()): String = {
+    madSend(new HttpGet(), "https://westus2.api.cognitive.microsoft.com/anomalydetector/" +
+      "v1.1-preview/multivariate/models", params)
+  }
+}
+
+trait MADUtils extends TestBase with AnomalyKey {
+
+  import spark.implicits._
+
+  lazy val df: DataFrame = Seq(
+    "https://mmlspark.blob.core.windows.net/datasets/sample_data_5_3000.zip"
+  ).toDF("source")
+
+  lazy val startTime: String = "2021-01-01T00:00:00Z"
+
+  lazy val endTime: String = "2021-01-02T12:00:00Z"
+}
+
+class MultivariateAnomalyModelSuite extends EstimatorFuzzing[MultivariateAnomalyModel] with MADUtils {
+
+  import MADUtils._
+
+  override def assertDFEq(df1: DataFrame, df2: DataFrame)(implicit eq: Equality[DataFrame]): Unit = {
+    def prep(df: DataFrame) = {
+      df.select("source")
+    }
+
+    super.assertDFEq(prep(df1), prep(df2))(eq)
+  }
+
+  def mad: MultivariateAnomalyModel = new MultivariateAnomalyModel()
+    .setSubscriptionKey(anomalyKey)
+    .setLocation("westus2")
```

Comment:
Do they have this service in eastus?

Reply:
Yes they do, but our existing one is in westus2 so I just use it.